landoapi/transplants.py

```diff
@@ -51,16 +65,115 @@
 CODE_FREEZE_OFFSET = "-0800"
 
 
+@dataclass
+class LandingAssessmentState:
+    """Encapsulates the state of a landing request for assessment.
+
+    Holds fields that are necessary to assess if a landing is blocked,
+    but are not necessary to run lints on the entire stack.
+    """
+
+    auth0_user: A0User
+    revision_path: list[str]
+    revision_to_diff_id: dict[str, int]
+    to_land: list[tuple[dict, dict]]
+
+    # `landing_repo` is set in the single landing repo check.
+    landing_repo: Optional[Repo] = None
+
+    @classmethod
+    def from_landing_path(
+        cls,
+        landing_path: list[tuple[int, int]],
+        stack_data: RevisionData,
+        auth0_user: A0User,
+    ) -> LandingAssessmentState:
+        landing_path_phid = convert_path_id_to_phid(landing_path, stack_data)
+
+        to_land = [stack_data.revisions[r_phid] for r_phid, _ in landing_path_phid]
+        to_land = [
+            (
+                revision,
+                stack_data.diffs[
+                    PhabricatorClient.expect(revision, "fields", "diffPHID")
+                ],
+            )
+            for revision in to_land
+        ]
+
+        revision_path = []
+        revision_to_diff_id = {}
+        for revision_phid, diff_id in landing_path_phid:
+            revision_path.append(revision_phid)
+            revision_to_diff_id[revision_phid] = diff_id
+
+        return LandingAssessmentState(
+            to_land=to_land,
+            revision_path=revision_path,
+            revision_to_diff_id=revision_to_diff_id,
+            auth0_user=auth0_user,
+        )
+
+
+@dataclass
+class TransplantAssessmentState:
+    """Handles the state of a Transplant for assessment.
```

Comment:
Docstring should describe what is meant by a "state of a Transplant" (also note that capitalizing "Transplant" is a little confusing here, because Transplant used to be a legacy model that is no longer used).